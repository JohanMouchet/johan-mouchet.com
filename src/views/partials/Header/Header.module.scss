@use "sass:map";
@import "@/assets/styles/settings/settings.module";
@import "@/assets/styles/tools/tools.module";

.p-header {
  $root: &;
  $skew-angle: 10deg;

  position: relative;
  width: 100%;
  min-height: 400px;
  overflow: hidden;
  text-shadow: 1px 1px 1px rgba($s-color-black, 0.3);
  background: map.get($s-backgrounds, dots),
    url("#{map.get($s-paths, theme)}header.jpg") left center/cover no-repeat
      $s-color-primary;

  &::before {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    content: "";
    background-image: linear-gradient(
      transparent,
      rgba($s-color-primary-darker, 0.9) 90%
    );
  }

  & > * {
    position: relative;
  }

  &__inner {
    display: flex;
    flex-wrap: wrap;
    align-content: center;
    min-height: inherit;
  }

  /* Heading */
  &__heading {
    display: flex;
    align-items: center;
    width: 100%;
    margin-bottom: 0;
    font-family: $s-font-family-display;
    font-size: t-modular-scale(4);
    font-weight: 400;
    color: $s-color-white;
    text-transform: uppercase;
    transition: font-size 0.5s;
    transform: skew(-$skew-angle);

    @media (min-width: map.get($s-breakpoints, sm)) {
      font-size: t-modular-scale(5);
    }

    @media (min-width: map.get($s-breakpoints, md)) {
      font-size: t-modular-scale(7);
    }

    @media (min-width: map.get($s-breakpoints, lg)) {
      font-size: t-modular-scale(9);
    }
  }

  /* Line */
  &__line {
    width: 2px;
    height: 2.5em;
    margin-right: 3px;
    margin-left: 3px;
    background: currentcolor;
    animation: line-animation 0.5s ease-out both;

    @keyframes line-animation {
      from {
        transform: scaleY(0);
      }
    }
  }

  /* Halfs */
  &__half {
    flex-basis: 50%;
    max-width: 50%;
    overflow: hidden;
  }

  &__half--left {
    padding-right: 0.2em;
    margin-bottom: 0.65em;
    text-align: right;

    #{$root}__text {
      animation: left-slide-animation 0.5s 0.5s ease-out both;
      /* stylelint-disable max-nesting-depth */
      @keyframes left-slide-animation {
        0% {
          opacity: 0;
          transform: translateX(100%);
        }

        50% {
          opacity: 1;
        }

        100% {
          transform: translateX(0);
        }
      }
      /* stylelint-enable max-nesting-depth */
    }
  }

  &__half--right {
    padding-left: 0.2em;
    margin-top: 0.65em;

    #{$root}__text {
      animation: right-slide-animation 0.5s 0.5s ease-out both;
      /* stylelint-disable max-nesting-depth */
      @keyframes right-slide-animation {
        0% {
          opacity: 0;
          transform: translateX(-100%);
        }

        50% {
          opacity: 1;
        }

        100% {
          transform: translateX(0);
        }
      }
      /* stylelint-enable max-nesting-depth */
    }
  }

  /* Text */
  &__text-wrapper {
    display: inline-block;
    transform: skew($skew-angle);
  }

  &__text {
    display: inline-block;
  }

  /* Tagline */
  &__tagline {
    width: 100%;
    margin: t-vertical-rhythm(2) t-vertical-rhythm(1) t-vertical-rhythm(1);
    font-family: $s-font-family-mono;
    font-size: t-modular-scale(-1);
    font-weight: 400;
    color: $s-color-secondary;
    text-align: center;
    text-transform: uppercase;
    letter-spacing: 0.4em;
    animation: tagline-animation 0.4s 1s ease-out both;

    @keyframes tagline-animation {
      0% {
        opacity: 0;
        transform: translateY(100%);
      }

      50% {
        opacity: 1;
      }

      100% {
        transform: translateY(0);
      }
    }
  }

  &__tagline-breaker {
    @media (min-width: map.get($s-breakpoints, md)) {
      display: none;
    }
  }
}
