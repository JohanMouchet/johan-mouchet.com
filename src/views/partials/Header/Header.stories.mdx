import { Canvas, Meta, Story } from "@storybook/addon-docs";
import { text } from "@storybook/addon-knobs";
import Header from ".";

<Meta title="Partials/Header" component={Header} />

# Header

<Canvas>
  <Story name="default">
    <Header
      firstName={text("firstName")}
      lastName={text("lastName")}
      tagline={text("tagline")}
    />
  </Story>
</Canvas>
