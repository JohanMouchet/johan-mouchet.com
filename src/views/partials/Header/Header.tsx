import * as React from "react";
import { PROFILE } from "constants/profile";
import parse from "utils/parse";
import "./Header.scss";

type Props = {
  firstName?: string;
  lastName?: string;
  tagline?: string;
};

const Header = ({
  firstName = PROFILE.firstName,
  lastName = PROFILE.lastName,
  tagline = /*html*/ `Never stop exploring,<br class="p-header__tagline-breaker" /> never stop creating`,
}: Props) => (
  <header className="p-header">
    <div className="p-header__inner">
      <h1 className="p-header__heading">
        <span className="p-header__half p-header__half--left">
          <span className="p-header__text-wrapper">
            <span className="p-header__text">{firstName}</span>
          </span>
        </span>

        <span className="p-header__line"></span>

        <span className="p-header__half p-header__half--right">
          <span className="p-header__text-wrapper">
            <span className="p-header__text">{lastName}</span>
          </span>
        </span>
      </h1>

      <h2 className="p-header__tagline">{parse(tagline)}</h2>
    </div>
  </header>
);

export default Header;
