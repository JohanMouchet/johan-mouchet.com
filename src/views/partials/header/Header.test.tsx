import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import { Header } from "./Header";

it("Renders <Header /> without crashing", () => {
  expect(
    render(
      <Header firstName="Firstname" lastName="Lastname" tagline="Tagline" />
    ).container.firstChild
  ).toMatchSnapshot();
});
