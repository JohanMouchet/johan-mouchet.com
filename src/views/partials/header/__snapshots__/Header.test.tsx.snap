// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders <Header /> without crashing 1`] = `
<header
  class="_p-header_8c45c1"
>
  <h1
    class="_p-header__heading_8c45c1"
  >
    Firstname
     
    Lastname
  </h1>
  <span
    class="_p-header__tagline_8c45c1"
  >
    Tagline
  </span>
</header>
`;
